p2p/p2p_test.go

```diff
@@ -30,51 +43,56 @@ func TestMain(m *testing.M) {
 		panic(fmt.Sprintf("logger initialization failed: %s", err))
 	}
 	globalData.Log = logger.New("nodes")
-	os.Exit(m.Run())
+	rc := m.Run()
+	_ = os.Remove(theLogFile)
+	os.Exit(rc)
 }
+
 func TestIDMarshalUnmarshal(t *testing.T) {
-	conf, err := mockConfiguration("server", 12136)
-	assert.NoError(t, err, "generate mock data error")
+	conf, err := mockConfiguration("server", unusedPortForTesting)
+	_ = assert.NoError(t, err, "generate mock data error") || die(t)
 	secretKey, err := util.DecodePrivKeyFromHex(conf.SecretKey)
-	assert.NoError(t, err, "Decode Hex Key Error")
+	_ = assert.NoError(t, err, "Decode Hex Key Error") || die(t)
 	id, err := peerlib.IDFromPrivateKey(secretKey)
-	assert.NoError(t, err, "IDFromSecretKey Error")
+	_ = assert.NoError(t, err, "IDFromSecretKey Error") || die(t)
 	mID, err := id.Marshal()
-	assert.NoError(t, err, "ID Marshal Error:")
+	_ = assert.NoError(t, err, "ID Marshal Error:") || die(t)
 	id2, err := peerlib.IDFromBytes(mID)
-	assert.NoError(t, err, "not a valid id bytes")
-	assert.Equal(t, id.String(), id2.String(), fmt.Sprintf("Convert ID fail! id:%v", id2.ShortString()))
+	_ = assert.NoError(t, err, "not a valid id bytes") || die(t)
+	assert.Equal(t, id.String(), id2.String(), fmt.Sprintf("Convert ID fail! id: %v", id2.ShortString()))
 
 }
 
 func TestNewP2P(t *testing.T) {
-	config, err := mockConfiguration("server", 22136)
-	assert.NoError(t, err, "mockdata generate error")
+	config, err := mockConfiguration("server", unusedPortForTesting)
+	_ = assert.NoError(t, err, "mockdata generate error") || die(t)
 	err = Initialise(config, "v1.0.0", false)
-	assert.NoError(t, err, "P2P  initialized error")
+	_ = assert.NoError(t, err, "P2P  initialized error") || die(t)
 	Finalise()
 }
 
 func TestListen(t *testing.T) {
-	config, err := mockConfiguration("server", 22136)
-	assert.NoError(t, err, "mockdata generate error")
+	config, err := mockConfiguration("server", unusedPortForTesting)
+	_ = assert.NoError(t, err, "mockdata generate error") || die(t)
 	n1 := Node{}
 	n1.Log = logger.New("p2p")
-	n1.Setup(config, "p2p-v1", false)
+	err = n1.Setup(config, "p2p-v1", false)
+	_ = assert.NoError(t, err, fmt.Sprintf("node setup error: %s", err)) || die(t)
 	conn, err := net.Dial("tcp", "127.0.0.1:"+strconv.Itoa(config.Port))
-	assert.NoError(t, err, fmt.Sprintf("listen error : %v", err))
+	_ = assert.NoError(t, err, fmt.Sprintf("listen error: %s", err)) || die(t)
 	conn.Close()
 	n1.Host.Close()
 }
 
 func TestIsTheSameNode(t *testing.T) {
-	config, err := mockConfiguration("server", 22136)
-	assert.NoError(t, err, "mockdata generate error")
```

Comment:
For your reference, some tests is about testing internal function/method of a package. There's suggestion about test methodology to test only exported instead of internal.

Reply:
I this I will leave the test in for now